### PR TITLE
Fix #7323: Fix missing PCP check for path dependent type

### LINKED ISSUE #7323: Pickling error with quoted path dependent types
```scala
trait Foo { type X }
def f[T](given scala.quoted.Type[T]) = ???
def g(m: Foo) = f[m.X]
```

Says:

<details><summary>Crash</summary>

```scala
exception occurred while compiling ../pg/i1.scala
java.lang.AssertionError: assertion failed: unresolved symbols: value m(line 2) when pickling ../pg/i1.scala while compiling ../pg/i1.scala
Exception in thread "main" java.lang.AssertionError: assertion failed: unresolved symbols: value m(line 2) when pickling ../pg/i1.scala
	at dotty.DottyPredef$.assertFail(DottyPredef.scala:17)
	at dotty.tools.dotc.core.tasty.TreePickler.pickle(TreePickler.scala:697)
	at dotty.tools.dotc.core.quoted.PickledQuotes$.pickle(PickledQuotes.scala:101)
	at dotty.tools.dotc.core.quoted.PickledQuotes$.pickleQuote(PickledQuotes.scala:33)
	at dotty.tools.dotc.transform.ReifyQuotes$QuoteReifier.pickleAsTasty$1(ReifyQuotes.scala:236)
	at dotty.tools.dotc.transform.ReifyQuotes$QuoteReifier.pickledQuote(ReifyQuotes.scala:244)
	at dotty.tools.dotc.transform.ReifyQuotes$QuoteReifier.transformQuotation(ReifyQuotes.scala:194)
	at dotty.tools.dotc.transform.TreeMapWithStages.transform(TreeMapWithStages.scala:93)
	at dotty.tools.dotc.transform.ReifyQuotes$QuoteReifier.transform(ReifyQuotes.scala:415)
	at dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform$$anonfun$2(Trees.scala:1352)
	at scala.collection.immutable.List.mapConserve(List.scala:444)
	at dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1352)
	at dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1262)
	at dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:117)
	at dotty.tools.dotc.transform.TreeMapWithStages.mapOverTree$1(TreeMapWithStages.scala:76)
	at dotty.tools.dotc.transform.TreeMapWithStages.transform(TreeMapWithStages.scala:126)
	at dotty.tools.dotc.transform.ReifyQuotes$QuoteReifier.transform(ReifyQuotes.scala:415)
	at dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:97)
	at dotty.tools.dotc.transform.TreeMapWithStages.mapOverTree$1(TreeMapWithStages.scala:76)
	at dotty.tools.dotc.transform.TreeMapWithStages.transform(TreeMapWithStages.scala:126)
	at dotty.tools.dotc.transform.ReifyQuotes$QuoteReifier.transform(ReifyQuotes.scala:415)
	at dotty.tools.dotc.ast.TreeMapWithImplicits.traverse$1(TreeMapWithImplicits.scala:51)
	at dotty.tools.dotc.ast.TreeMapWithImplicits.transformStats(TreeMapWithImplicits.scala:58)
	at dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:108)
	at dotty.tools.dotc.transform.TreeMapWithStages.mapOverTree$1(TreeMapWithStages.scala:76)
	at dotty.tools.dotc.transform.TreeMapWithStages.transform(TreeMapWithStages.scala:126)
	at dotty.tools.dotc.transform.ReifyQuotes$QuoteReifier.transform(ReifyQuotes.scala:415)
	at dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1332)
	at dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:101)
	at dotty.tools.dotc.transform.TreeMapWithStages.mapOverTree$1(TreeMapWithStages.scala:76)
	at dotty.tools.dotc.transform.TreeMapWithStages.transform(TreeMapWithStages.scala:126)
	at dotty.tools.dotc.transform.ReifyQuotes$QuoteReifier.transform(ReifyQuotes.scala:415)
	at dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform$$anonfun$2(Trees.scala:1352)
	at scala.collection.immutable.List.mapConserve(List.scala:444)
	at dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1352)
	at dotty.tools.dotc.ast.Trees$Instance$TreeMap.transformStats(Trees.scala:1350)
	at dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1338)
	at dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:101)
	at dotty.tools.dotc.transform.TreeMapWithStages.mapOverTree$1(TreeMapWithStages.scala:76)
	at dotty.tools.dotc.transform.TreeMapWithStages.transform(TreeMapWithStages.scala:126)
	at dotty.tools.dotc.transform.ReifyQuotes$QuoteReifier.transform(ReifyQuotes.scala:415)
	at dotty.tools.dotc.transform.ReifyQuotes$$anon$1.transform(ReifyQuotes.scala:93)
	at dotty.tools.dotc.transform.MacroTransform.run(MacroTransform.scala:21)
	at dotty.tools.dotc.transform.ReifyQuotes.run(ReifyQuotes.scala:89)
	at dotty.tools.dotc.core.Phases$Phase.runOn$$anonfun$1(Phases.scala:315)
	at scala.collection.immutable.List.map(List.scala:219)
	at dotty.tools.dotc.core.Phases$Phase.runOn(Phases.scala:316)
	at dotty.tools.dotc.Run.runPhases$4$$anonfun$4(Run.scala:159)
	at dotty.runtime.function.JProcedure1.apply(JProcedure1.java:15)
	at dotty.runtime.function.JProcedure1.apply(JProcedure1.java:10)
	at scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1323)
	at dotty.tools.dotc.Run.runPhases$5(Run.scala:169)
	at dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:177)
	at dotty.runtime.function.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:67)
	at dotty.tools.dotc.Run.compileUnits(Run.scala:184)
	at dotty.tools.dotc.Run.compileSources(Run.scala:121)
	at dotty.tools.dotc.Run.compile(Run.scala:104)
	at dotty.tools.dotc.Driver.doCompile(Driver.scala:35)
	at dotty.tools.dotc.Driver.process(Driver.scala:178)
	at dotty.tools.dotc.Driver.process(Driver.scala:147)
	at dotty.tools.dotc.Driver.process(Driver.scala:159)
	at dotty.tools.dotc.Driver.main(Driver.scala:186)
	at dotty.tools.dotc.Main.main(Main.scala)
[error] Nonzero exit code returned from runner: 1
```
</details>

The following works:

```scala
trait Foo { type X }
def f[T](given scala.quoted.Type[T]) = ???
def g(m: Foo)(given scala.quoted.Type[m.X]) = f[m.X]
```

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Note that the pseudo fix for #7048 causes more alarming bugs.

**Comment by nicolasstucki** [comment]
It is better to reopen the feature request in #7048 than have a bug that crashes the compiler.

File: compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala [hunk 1]
```diff
@@ -119,8 +119,7 @@ class PCPCheckAndHeal(@constructorOnly ictx: Context) extends TreeMapWithStages(
           // Replace it with a properly encoded type splice. This is the normal for expected for type splices.
           tp.prefix.select(tpnme.splice)
         case tp: NamedType =>
-          if (tp.prefix.isInstanceOf[TermRef] && tp.prefix.isStable) tp
-          else checkSymLevel(tp.symbol, tp, pos) match {
+          checkSymLevel(tp.symbol, tp, pos) match {
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala]: What if we just change change test to `tp.symbol.isStatic`?

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala]: The checkSymLevel already check that

**Pull request merged**